cpp/test/sg/rf_treelite_test.cu

```diff
@@ -270,10 +274,41 @@ class RfTreeliteTestClf : public RfTreeliteTestCommon<T, L> {
     // #class for multi-class classification
     this->task_category = 2;
 
-    // Populate labels randomly.
+    float *weight, *noise, *temp_label_d;
+    std::vector<float> temp_label_h;
+
+    allocate(weight, this->params.n_cols);
+    allocate(noise, this->params.n_rows);
+    allocate(temp_label_d, this->params.n_rows);
+
+    Random::Rng r(1234ULL);
+
+    r.uniform(weight, this->params.n_cols, T(0.0), T(1.0), this->stream);
+    r.uniform(noise, this->params.n_rows, T(0.0), T(10.0), this->stream);
+
+    LinAlg::gemm<float, float, float, cutlass::Shape<8, 128, 128>>(
+      CUBLAS_OP_N, CUBLAS_OP_N, this->params.n_rows, this->params.n_cols, 1,
+      1.f, this->data_d, this->params.n_rows, weight, this->params.n_cols, 1.f,
+      noise, this->params.n_rows, temp_label_d, this->stream);
+    CUDA_CHECK(cudaStreamSynchronize(this->stream));
```

Comment:
`cudaStreamSynchronize()` is not needed here, the GPU operations on the same stream run sequentially.